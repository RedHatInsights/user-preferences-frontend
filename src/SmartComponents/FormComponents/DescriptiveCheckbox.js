import React from 'react';
import { Checkbox } from '@patternfly/react-core';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './descriptiveCheckbox.scss';

// eslint-disable-next-line no-unused-vars
const DescriptiveCheckbox = ({ name, label, description, isDanger, FieldProvider, formOptions, ...rest }) => (
    <Checkbox
        { ...rest }
        id={ `descriptive-checkbox-${name}` }
        className="pref-c__descriptive-checkbox"
        label={ <span className={ classNames('pref-c__checkbox-label', { 'pref-c__checkbox-label-error': isDanger }) }>{label}</span> }
        description={ <span className="pref-c__checkbox-description">{description}</span> }
    />
);

DescriptiveCheckbox.propTypes = {
    FieldProvider: PropTypes.any.apply,
    formOptions: PropTypes.any,
    name: PropTypes.string,
    label: PropTypes.string,
    description: PropTypes.string,
    isDanger: PropTypes.bool
};

DescriptiveCheckbox.defaultProps = {
    name: '',
    label: '',
    isDanger: false
};

export default DescriptiveCheckbox;
