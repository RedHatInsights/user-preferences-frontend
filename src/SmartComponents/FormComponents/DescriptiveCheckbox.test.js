import React from 'react';
import DescriptiveCheckbox from './DescriptiveCheckbox';
import { Form, RendererContext } from '@data-driven-forms/react-form-renderer';
import { fireEvent, getByRole, render } from '@testing-library/react';

describe('Descriptive checkbox tests', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{
              formOptions: {
                internalRegisterField: () => undefined,
                internalUnRegisterField: () => undefined,
              },
            }}
          >
            <DescriptiveCheckbox label="test label" clearedValue {...props} />
          </RendererContext.Provider>
        )}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  it('should call beforeOnChange correctly', () => {
    const beforeOnChange = jest.fn();
    const { container } = render(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{
              formOptions: {
                internalRegisterField: () => undefined,
                internalUnRegisterField: () => undefined,
              },
            }}
          >
            <DescriptiveCheckbox
              label="test label"
              group="testGroup"
              section="testSection"
              clearedValue
              {...props}
              beforeOnChange={beforeOnChange}
              isGlobal
            />
          </RendererContext.Provider>
        )}
      </Form>
    );
    fireEvent.click(getByRole(container, 'checkbox'));
    expect(beforeOnChange).toHaveBeenCalledTimes(1);
    expect(beforeOnChange).toHaveBeenNthCalledWith(
      1,
      true,
      false,
      expect.anything(),
      'testGroup',
      'testSection',
      ''
    );
  });
});
