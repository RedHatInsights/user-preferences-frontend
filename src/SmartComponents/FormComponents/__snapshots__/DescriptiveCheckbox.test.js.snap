// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Descriptive checkbox tests should render correctly 1`] = `
<div>
  <div
    class="pf-v6-c-check pref-c-descriptive-checkbox"
  >
    <input
      aria-invalid="false"
      checked=""
      class="pf-v6-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-1"
      data-ouia-component-type="PF6/Checkbox"
      data-ouia-safe="true"
      data-type="descriptive-checkbox"
      id="descriptive-checkbox-"
      name=""
      type="checkbox"
      value=""
    />
    <label
      class="pf-v6-c-check__label"
      for="descriptive-checkbox-"
    >
      test label
    </label>
    <span
      class="pf-v6-c-check__description"
    >
      <div />
    </span>
  </div>
</div>
`;
